src/hotspot/share/opto/vectornode.cpp

```diff
@@ -150,9 +150,21 @@ int VectorNode::opcode(int sopc, BasicType bt) {
   case Op_RoundDoubleMode:
     return (bt == T_DOUBLE ? Op_RoundDoubleModeV : 0);
   case Op_RotateLeft:
-    return (is_integral_type(bt) ? Op_RotateLeftV : 0);
+    switch (bt) {
+    case T_INT:
+    case T_LONG: return Op_RotateLeftV;
+    default:     return 0; // RotateLeftV for byte, short values produces incorrect Java result.
+                           // Because java code should convert a byte, short value into int value,
+                           // and then do RotateI.
```

Comment:
I‘m afraid this will influence the VectorAPI intrinsification for `ByteVector `and `ShortVector`. Did you test the benchmarks for these two APIs with the subword type?

Reply:
I have tested `hotspot_vector_1` and `jdk_vector`, all pass.

Reply:
> Do you mean the benchmarks? Currently the vectorapi benchmarks only exist on panama-vector, please see: https://github.com/openjdk/panama-vector/tree/vectorIntrinsics/test/micro/org/openjdk/bench/jdk/incubator/vector/operation

I have tried to make images on branch vectorIntrinsics of panama-vector, but get an error.

```
warning: unknown enum constant Feature.FOREIGN
warning: unknown enum constant Feature.FOREIGN
warning: unknown enum constant Feature.FOREIGN
error: warnings found and -Werror specified
1 error
7 warnings
make[3]: *** [/home/wx/wangxue/panama-vector/build/linux-x86_64-server-release/buildtools/interim_langtools_modules/java.compiler.interim/_the.BUILD_java.compiler.interim_batch] Error 1
make[2]: *** [interim-langtools] Error 2
```

Reply:
I‘m sorry that I just build the latest vectorIntrinsics branch and didn't met the error. Did you just cherry-pick your patch on it? Thanks!

Reply:
```
-      CFLAGS_WARNINGS_ARE_ERRORS="-Werror"
+      CFLAGS_WARNINGS_ARE_ERRORS=""
...
-  JAVA_WARNINGS_ARE_ERRORS ?= -Werror
+  JAVA_WARNINGS_ARE_ERRORS ?= 
...
```
with the dirty diff, I have tested Byte128Vector.
Yes, the Score about rotate is lower than before my patch.
Could you give me some suggestions?

Reply:
Since `"VectorNode::implemented"` is only used for auto-vect, I think we can do the check in `"VectorNode::is_vector_rotate_supported" `and return false for byte and short type.

Reply:
Or you can try to add the right support for byte/short for some cases like https://github.com/openjdk/jdk/pull/7979